dolphin/api/v1/access_info.py

```diff
@@ -37,8 +42,36 @@ def show(self, req, id):
 
         return self._view_builder.show(access_info)
 
+    @validation.schema(schema_access_info.update)
+    def update(self, req, id, body):
+        """Update storage access information."""
+        ctxt = req.environ.get('dolphin.context')
+        # Get existing access_info and storage from DB
+        try:
+            access_info = db.access_info_get(ctxt, id)
+            access_info.update(body)
+            storage = self.driver_api.discover_storage(ctxt, access_info)
+            storage_present = db.storage_get(ctxt, id)
+            if storage['serial_number'] != storage_present.serial_number:
+                reason = (_("Existing storage serial Number is not matching \
+                with th new storage serial number: '%s'  ") % storage['serial_number'])
+                raise exception.StorageSerialNumberMismatch(reason=reason)
+            db.storage_update(ctxt, id, storage)
+            access_info['password'] = cryptor.encode(
+                access_info['password'])
+            db.access_info_update(ctxt, id, access_info)
+        except (exception.InvalidCredential,
+                exception.StorageDriverNotFound,
+                exception.AccessInfoNotFound,
+                exception.StorageNotFound,
+                exception.StorageSerialNumberMismatch) as e:
+            raise exc.HTTPBadRequest(explanation=e.message)
+        except Exception as e:
+            msg = _('Failed to to update  access info: {0}'.format(e))
+            LOG.error(msg)
+            raise exc.HTTPBadRequest(explanation=msg)
+        return storage_view.build_storage(storage)
```

Comment:
need to return access_info here, not storage

Reply:
done